Add enable and disable state to service group member bindings...

Hollas99 opened the pull request:
Many users add servers to their setup first in a disabled state. This change adds support for both
state: enabled | disabled

`servicegroup_servicegroupmember_binding` and `gslbservicegroup_gslbservicegroupmember_binding`

using either enabled, disabled with update the resource but also set the state parameter for nitro api, when using the state as present, the admin state will not be adjusted, unless its a new resource then the default is used as expected.

Hollas99 commented:
This is a partial solution to #422, and does not currently cover the suggested 'monitor_name' name change suggestion

shivashankar-vaddepally commented:
HI @Hollas99 
I have permitted the PR for the builds. Can you please look into it while I review the PR? 

Thanks

Hollas99 commented:
Hey @shivashankar-vaddepally, I've rebased and fixed what was causing the tests to fail if you could please rerun them

shivashankar-vaddepally commented:
Hi @Hollas99 
Appreciate the effort to resolve issues causing tests to fail. Can you please look into the comments on changes too?

Thanks

Hollas99 commented:
> Hi @Hollas99 Appreciate the effort to resolve issues causing tests to fail. Can you please look into the comments on changes too?
> 
> Thanks

@shivashankar-vaddepally - Sorry but I dont see any comments left on the code at all.

shivashankar-vaddepally commented:
Hi @Hollas99 
Please check the ansible-test checks which are failing and also I have given comments in the PR itself as in attached below. Not sure why you cant see those comments.  Let me check .
<img width="560" alt="comments" src="https://github.com/user-attachments/assets/24f2d8e1-9c62-4c6b-b30e-9e10bf3ecf09" />

Hollas99 commented:
> Contributor

Hi @shivashankar-vaddepally , without this code, the resource will not be created in a disabled state if provided - always enabled.

The other way I can think of that would not affect other resources is I remove this code, and

add `state` to the `add_payload_keys` in the NITRO_RESOURCE_MAP

Please advise, Thanks.

shivashankar-vaddepally commented:
Hi @Hollas99 
I assume you're currently reviewing the forked repository where the comments appear to be missing. Could you please take a look at the original NetScaler repository instead? That's the one I'm actively working on.

Thanks!

shivashankar-vaddepally commented:
Hi @Hollas99 
We see that all checks are successfully executed. 

I request you to look into the other 2 comments which are pasted, so that this can be closed asap. 


thanks

Hollas99 commented:
Hey @shivashankar-vaddepally 

I reorganised a lot of the code - I've moved a lot of it and rewritten some into `create_or_update`, this should align to all the comments you have a left, and I feel the code is now organised much better. Let me know what you think. 

Thanks in advance.

..also, I am unable to see your code comments as they are 'pending' - I believe the maintainer needs to approve them before they are shown.

shivashankar-vaddepally commented:
HI @Hollas99 
I reviewed it, it looks fine.

Thanks
